azblob/parsing_urls.go

```diff
@@ -124,6 +124,10 @@ func (up BlobURLParts) URL() url.URL {
 
 	rawQuery := up.UnparsedParams
 
+	if up.Snapshot == "" && !up.SAS.snapshotTime.IsZero() {
```

Comment:
Please add a comment here to clarify the intent. 

If I understood it correctly, if the SAS targets a snapshot, and no snapshot ID was specified, we help the user by specifying it.